### PR TITLE
Remove event aliases if none exist.

### PR DESCRIPTION
**Pull request by SSlayer9**
https://www.smogon.com/forums/threads/bug-reports-v4-read-original-post-before-posting.3663703/post-8488586 Might not work but it's worth a shot.

I literally just looked at all the html bits and edited it, i still don't know how events are stored or anything, please feel free to edit if I'm being stupid, UwU.

### CODE REVIEW

File: server/chat-plugins/room-events.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ function formatEvent(event: {eventName: string, date: string, desc: string, star
 	}
 	let ret = `<tr title="${explanation}">`;
 	ret += Chat.html`<td>${event.eventName}</td>`;
-	ret += Chat.html`<td>${event.aliases ? event.aliases.join(", ") : ""}</td>`;
+	ret += event.aliases ? Chat.html`<td>${event.aliases.join(", ")}</td>` : ""
```
Comment (AnnikaCodes) [inline server/chat-plugins/room-events.ts:21]:
```suggestion
	ret += event.aliases.length ? Chat.html`<td>${event.aliases.join(", ")}</td>` : "";
```

Comment (AnnikaCodes) [reply server/chat-plugins/room-events.ts:21]: Also, you need to figure out a way to include empty `<td>` tags when displaying all events if one event has alias(es) and others do not - I'll leave that for you to figure out (maybe pass an argument to `formatEvent()`?). Currently, your code doesn't work when showing all aliases.

File: server/chat-commands/info.ts [hunk 2]
```diff
@@ -2463,7 +2463,6 @@ export const commands: ChatCommands = {
 	},
 	showimagehelp: [`/showimage [url], [width], [height] - Show an image. Any CSS units may be used for the width or height (default: px). If width and height aren't provided, automatically scale the image to fit in chat. Requires: # & ~`],
 
-	'!pi': true,
```
Comment (AnnikaCodes) [inline server/chat-commands/info.ts:2466]: This should be in a separate pull request. `!pi` has nothing to do with room events.

File: server/chat-commands/info.ts [hunk 3]
```diff
@@ -2462,8 +2462,8 @@ export const commands: ChatCommands = {
 		});
 	},
 	showimagehelp: [`/showimage [url], [width], [height] - Show an image. Any CSS units may be used for the width or height (default: px). If width and height aren't provided, automatically scale the image to fit in chat. Requires: # & ~`],
-
-	'!pi': true,
+	
+	"!pi": true,
```
Comment (AnnikaCodes) [inline server/chat-commands/info.ts:2466]:
```suggestion

	'!pi': true,
```

File: server/chat-plugins/room-events.ts [hunk 4]
```diff
@@ -53,7 +53,7 @@ export const commands: ChatCommands = {
 			}
 			if (!this.runBroadcast()) return;
 			let buff = '<table border="1" cellspacing="0" cellpadding="3">';
-			buff += '<th>Event Name:</th><th>Event Aliases:</th><th>Event Description:</th><th>Event Date:</th>';
+			buff += '<th>Event Name:</th>' + getAllAliases(room).length ? '<th>Event Aliases:</th>' : "" + '<th>Event Description:</th><th>Event Date:</th>';
```
Comment (Zarel) [inline server/chat-plugins/room-events.ts:56]:
```suggestion
			buff += `<th>Event Name:</th>${getAllAliases(room).length ? `<th>Event Aliases:</th>` : ``}<th>Event Description:</th><th>Event Date:</th>`;
```

File: server/chat-plugins/room-events.ts [hunk 5]
```diff
@@ -18,7 +18,7 @@ function formatEvent(event: {eventName: string, date: string, desc: string, star
 	}
 	let ret = `<tr title="${explanation}">`;
 	ret += Chat.html`<td>${event.eventName}</td>`;
-	ret += Chat.html`<td>${event.aliases ? event.aliases.join(", ") : ""}</td>`;
+	ret += event.aliases.length ? Chat.html`<td>${event.aliases.join(", ")}</td>` : "";
```
Comment (Zarel) [inline server/chat-plugins/room-events.ts:21]:
```suggestion
	ret += event.aliases.length ? Chat.html`<td>${event.aliases.join(", ")}</td>` : ``;
```

**Comment by Zarel** [comment]
Thanks!
